### PR TITLE
feat: [PIN-2439] Added CID to stacktraces and uniformation of logging

### PR DESCRIPTION
**Pull request by TonioGela**
As you can see, I had to change a few things to make it work. 

**This change is breaking and requires patching every microservices.** 

I time-capped the modification and thus avoided modifying and refactoring the `OpenApiUtils` functions that check and log for API contract mismatch calls.

### CODE REVIEW

File: jwt/src/main/scala/it/pagopa/interop/commons/jwt/service/JWTReader.scala
Comment (galales) [inline jwt/src/main/scala/it/pagopa/interop/commons/jwt/service/JWTReader.scala]: may I suggest to handle the nullability right here to avoid carrying around a possible `null`?
```suggestion
    organizationId <- Try(Option(claims.getStringClaim(ORGANIZATION_ID_CLAIM)).getOrElse(""))
```

File: jwt/src/main/scala/it/pagopa/interop/commons/jwt/service/JWTReader.scala
Comment (galales) [inline jwt/src/main/scala/it/pagopa/interop/commons/jwt/service/JWTReader.scala:27]: just for confirmation, is this the only breaking change of the PR?

**Review by galales — COMMENTED** [review]
may I ask you to not merge this until the error handling PRs are merged? 🙏

File: jwt/src/main/scala/it/pagopa/interop/commons/jwt/service/JWTReader.scala
Comment (TonioGela) [reply jwt/src/main/scala/it/pagopa/interop/commons/jwt/service/JWTReader.scala:27]: nope, `logHttp` changed signature too, and that thing should be changed in templates

File: jwt/src/main/scala/it/pagopa/interop/commons/jwt/service/JWTReader.scala
Comment (TonioGela) [reply jwt/src/main/scala/it/pagopa/interop/commons/jwt/service/JWTReader.scala]: I totally can, but I have to follow the stack of calls to see whether it changes some behaviour downstream :D

Comment (TonioGela) [reply jwt/src/main/scala/it/pagopa/interop/commons/jwt/service/JWTReader.scala]: btw it's handled a few lines below

```scala
val orgId: List[(String, String)] = Option(organizationId).map(ORGANIZATION_ID_CLAIM -> _).toList
```

but it's not added to the context list at all if it's null. I can't tell whether changing this behaviour will break something downstream, I prefer to leave it as it is in this PR, it's a PR about logging after all. Do you agree?

Comment (galales) [reply jwt/src/main/scala/it/pagopa/interop/commons/jwt/service/JWTReader.scala]: Not sure I got your point, the `organizationId` would be handled the same way, the only other place where it is used is where you indicated, and it's still handled as an `Option`.
What I'm suggesting is just to wrap it as soon as it is obtained from the claim to avoid future changes to mistakenly use a `null` value

**Review by galales — APPROVED** [review]

**Pull request merged**
